router/router.go

```diff
@@ -535,6 +535,9 @@ func (self *Router) connectToControllerWithBackoff(addr *UpdatableAddress, local
 	}
 
 	operation := func() error {
```

Comment:
Still needs to be protected by lock. Might be simpler to just use a cmap.ConcurrentMap. Depends on if you're using the lock just for concurrent access or if you're also protecting a critical section.